src/main/java/com/rackspace/salus/telemetry/ambassador/services/MetricRouter.java

```diff
@@ -115,38 +112,57 @@ public void route(String tenantId, String envoyId,
+                .build()).collect(Collectors.toList());
+        metrics.addAll(nameTagValue.getFvaluesMap().entrySet().stream()
+            .map(entry -> Metric.newBuilder()
+                .setGroup(measurementName)
+                .setTimestamp(getProtoBufTimestamp(timestamp))
+                .setName(entry.getKey())
+                .setFloat(entry.getValue())
+                .putAllMetadata(tagsMap)
+                .build()).collect(Collectors.toList()));
+        metrics.addAll(nameTagValue.getSvaluesMap().entrySet().stream()
+            .map(entry -> Metric.newBuilder()
+                .setGroup(measurementName)
+                .setTimestamp(getProtoBufTimestamp(timestamp))
+                .setName(entry.getKey())
+                .setString(entry.getValue())
+                .putAllMetadata(tagsMap)
+                .build()).collect(Collectors.toList()));
+
+        final UniversalMetricFrame universalMetricFrame = UniversalMetricFrame.newBuilder()
+            .setAccountType(UniversalMetricFrame.AccountType.MANAGED_HOSTING)
+            .setTenantId(tenantId)
             .setDevice(resourceId)
-            .setDeviceMetadata(labels)
-            .setMonitoringSystem(MonitoringSystem.SALUS)
-            .setSystemMetadata(Collections.singletonMap("envoy_id", envoyId))
-            .setCollectionMetadata(tagsMap)
-            .setCollectionName(measurementName)
-            .setFvalues(nameTagValue.getFvaluesMap())
-            .setSvalues(nameTagValue.getSvaluesMap())
-            .setIvalues(Collections.emptyMap())
-            .setUnits(Collections.emptyMap())
+            .setDeviceName(nameTagValue.getName())
+            .putAllDeviceMetadata(labels)
+            .putAllSystemMetadata(Collections.singletonMap("envoy_id", envoyId))
+            .setMonitoringSystem(UniversalMetricFrame.MonitoringSystem.SALUS)
+            .addAllMetrics(metrics)
             .build();
 
-        final ByteArrayOutputStream out = new ByteArrayOutputStream();
         try {
-            final Schema schema = externalMetric.getSchema();
-            final JsonEncoder jsonEncoder = avroEncoderFactory.jsonEncoder(schema, out);
-
-            final SpecificDatumWriter<Object> datumWriter = new SpecificDatumWriter<>(schema);
-            datumWriter.write(externalMetric, jsonEncoder);
-            jsonEncoder.flush();
 
             metricsRouted.increment();
             kafkaEgress.send(Strings.join(List.of(tenantId, resourceId, measurementName), ','),
-                KafkaMessageType.METRIC, out.toString(StandardCharsets.UTF_8.name()));
+                KafkaMessageType.METRIC, JsonFormat.printer().print(universalMetricFrame));
 
-        } catch (IOException|NullPointerException e) {
-            log.warn("Failed to Avro encode avroMetric={} original={}", externalMetric, postedMetric, e);
+        } catch (IOException | NullPointerException e) {
+            log.warn("Failed to Avro encode avroMetric={} original={}", universalMetricFrame,
```

Comment:
Can get rid of Avro mention and also make generic with
```suggestion
            log.warn("Failed to encode metricFrame={} from={}", universalMetricFrame,
```